src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.Internal.cs

```diff
@@ -0,0 +1,141 @@
+        /// <param name="validationParameters">The <see cref="ValidationParameters"/> to be used for validating the token.</param>
+        /// <param name="callContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>An <see cref="ValidationResult{TResult}"/> that contains either the issuer that was validated or an error.</returns>
+        /// <remarks>An EXACT match is required.</remarks>
+        internal async Task<ValidationResult<ValidatedIssuer>> ValidateIssuerAsync(
+            string issuer,
+            SecurityToken securityToken,
+            ValidationParameters validationParameters,
+#pragma warning disable CA1801 // Review unused parameters
+            CallContext callContext,
+            CancellationToken cancellationToken)
+#pragma warning restore CA1801 // Review unused parameters
+        {
+            _ = issuer ?? throw LogHelper.LogArgumentNullException(nameof(issuer));
+            _ = securityToken ?? throw LogHelper.LogArgumentNullException(nameof(securityToken));
+            _ = validationParameters ?? throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+            string tenantId;
+
+            try
+            {
+                tenantId = GetTenantIdFromToken(securityToken);
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                return new IssuerValidationError(
+                    new MessageDetail(
+                        ex.Message,
+                        LogHelper.MarkAsNonPII(issuer)),
+                        ValidationFailureType.IssuerValidationFailed,
+                        typeof(SecurityTokenInvalidIssuerException),
+                        new StackFrame(true),
+                        issuer,
+                        ex);
+            }
+
+            if (string.IsNullOrWhiteSpace(tenantId))
+                return new IssuerValidationError(
+                    new MessageDetail(
+                        LogMessages.IDX40003,
+                        LogHelper.MarkAsNonPII(issuer)),
+                        ValidationFailureType.IssuerValidationFailed,
+                        typeof(SecurityTokenInvalidIssuerException),
+                        new StackFrame(true),
+                        issuer);
+
+            for (int i = 0; i < validationParameters.ValidIssuers.Count; i++)
+            {
+                if (IsValidIssuer(validationParameters.ValidIssuers[i], tenantId, issuer))
+                    return new ValidatedIssuer(issuer!, IssuerValidationSource.IssuerMatchedConfiguration);
+            }
+
+            try
+            {
+                var issuerVersion = GetTokenIssuerVersion(securityToken);
+                var effectiveConfigurationManager = GetEffectiveConfigurationManager(issuerVersion);
+
+                string aadIssuer = null;
```

Comment:
```suggestion
                string aadIssuer;
```

This gets unconditionally wrote to below.